### ISSUE TITLE
Support for arrow => functions

### ISSUE THREAD

**Issue by semmel** [yamadapc/jsdoctest#34]
```javascript
/**
 * @example
 *    divide(10, 10) // => 1
 */
const divide = (x, y) => x / y;
```
is not recognised by jsdoctest (_"0 passing"_) because [dox.js](https://github.com/tj/dox) detects a declaration:
```javascript
...
ctx: 
     { type: 'declaration',
       name: 'divide',
       value: '(x, y) =>'
...
```
This could be fixed easy by supplying a regular expression for ES6 arrow functions which would then change the context to `function`.
Right in the same way (with some [regular expressions](https://github.com/cjy37/dox/commit/98be32e44d855410054321e3afe69f32b4374bca) ) support for async/await can be added. As discussed here: #31.

Where should this happen? As a pull request into [dox.js](https://github.com/tj/dox) or by using the [dox api for context matching](https://github.com/tj/dox#extending-context-matching) in jsdoctest?
 
I'd prefer patching `getJsdoctests()` [here](https://github.com/yamadapc/jsdoctest/blob/ccd6cdbff8a4db98d20da8ac1868558a45913005/lib/index.js#L39).

**Comment by yamadapc** [comment]
Hm... Is the `value` incorrectly trimmed like that? I'd rather not use RegExp to parse JavaScript and have that be in `dox` or it's parsing lib, for sure.

If you're able to do it in a reliable way with RegExp I would gladly accept it though. I'm just not very confident about that solution.

**Comment by semmel** [comment]
Yeah you're probably right that all those RegExpes do not belong here. I am just not confident that a pull request would be accepted in [tj/dox](https://github.com/tj/dox) because the activity there is low.

However I am half-way through in upgrading [my fork of dox](https://github.com/semmel/dox) with async and arrow functions. It is not that difficult with the work @cjy37 has done but writing the missing tests takes time...

On the topic of async functions I'd really like to see the promise syntax `~>` `~!>` @jmatsushita has implemented in his [fork of jsdoctest](https://github.com/iilab/jsdoctest) to be supported. But I don't know if he's done more than this syntactic improvement that cannot be merged back to jsdoctest.
